index.ts

```diff
@@ -0,0 +1,552 @@
+// Exported native API for @smogon/data
+//
+// DISCLAIMER:
+//
+//   Not meant to replace Dex/BattleDex: consumers can delegate to this native API
+//   and put their own layer on top which can extend or alter the functionality.
+//
+// CONSIDERATIONS:
+//
+//    - Approach: async APIs are moved to top level so that data load can be done
+//      once and then the rest of the APIs aren't cluttered with `await` everywhere.
+//      Furthermore, individual pieces can be loaded (just `Abilities` vs. all data
+//      types) if desired.
+//    - `getByID` APIs are important for removing the `toID` call in `get` methods
+//      for when we already know through typechecking that we have an ID. String
+//      getters have been removed - callees can manually called `toID` first if they
+//      dont have an ID (which should be very rare unless its human-input).
+//    - Descriptions are UI only and can be large so aren't included in the Dex by
+//      default, but they may be loaded in bulk asynchronously for the desired
+//      languages.
+//    - Similarly, SpeciesDetails which are purely cosmetic or used by the Validator
+//      or for non-competitive reasons are offloaded. eg. learnset details are very
+//      large and not needed in many circumstances.
+//    - All other APIs are loaded together for a gen (+ mod) - it's rare a consumer
+//      cares *just* about Abilities or *just* Moves, and if they do and are in an
+//      environment where loading more is a concern they can implement their own
+//      splitting on top of the raw data files.
+//    - The aliases.json data is used internally (to implement getByID's) but is
+//      not exported.
```

Comment:
In the future I'd like to make this more useful for other purposes, but probably a post MVP concern

Reply:
What did you have in mind? I didn't really have any good public-facing ideas for it, but I wouldn't be against exposing it via some API.